v2/passwordless/serverless/with-aws-lambda/session-verification.mdx

```diff
@@ -11,57 +11,112 @@ import AppInfoForm from "/src/components/appInfoForm"
 import BackendSDKTabs from "/src/components/tabs/BackendSDKTabs";
 import TabItem from '@theme/TabItem';
 
-# 5. Session verification / Building your APIs
+# 3. Session verification / Building your APIs
 
-For this guide, we will assume that we want a new API Gateway endpoint `/user GET` invoked by the same lambda function and it returns the current session information.
+When building your own APIs, you may need to verify the session of the user before proceeding further. SuperTokens SDK exposes a `verifySession` function that can be utilized for this. In this guide, we will be creating a `/user` `GET` route that will return the current session information.
 
-## 1) Copy this snippet to your handler file
+## 1) Add `/user` `GET` route in your API Gateway
 
-We use the verify session function to get the session information.
+Create a `/user` resource and then `GET` method in your API Gateway. Configure the lambda integration and CORS just like we did [for the auth routes](./setup-api-gateway#3-attach-lambda-to-the-any-method-of-the-proxy-resource).
+
+## 2) Create a file in your lambda to handle the `/user` route.
 
 <BackendSDKTabs disableGolang>
 <TabItem value="nodejs">
 
-An example of this is [here](https://github.com/supertokens/supertokens-auth-react/blob/master/examples/with-aws-lambda/backend/user.js).
+An example of this is [here](https://github.com/supertokens/supertokens-auth-react/blob/master/examples/with-aws-lambda/backend/user.mjs).
 
-```tsx title="auth.ts"
+```tsx title="user.mjs"
 import supertokens from "supertokens-node";
 // @ts-ignore
-import { getBackendConfig } from "./config";
+import { getBackendConfig } from "./config.mjs";
 import { verifySession } from "supertokens-node/recipe/session/framework/awsLambda";
-import { SessionEvent } from "supertokens-node/framework/awsLambda";
 import middy from "@middy/core";
 import cors from "@middy/http-cors";
 
 supertokens.init(getBackendConfig());
 
-//highlight-start
-const handler = async (event: SessionEvent) => {
+// @ts-ignore
```

Comment:
can we avoid using ts-ignore here?

Reply:
No.. we get an "implicit any" error there. Can't really add a type there because it is an `.mjs` file.

Reply:
The issue with this is that the use of the session object in the function won't be type checked anymore. Maybe you can introduce some way of adding types to the code snippets, but then remove them during rendering? An example of this could be:
```
const lambdaHandler = async (event__HIDDEN_TYPE_START__: AWSEvent__HIDDEN_TYPE_END__) => {
    return {
        body: JSON.stringify({
            sessionHandle: event.session.getHandle(),
            userId: event.session.getUserId(),
            accessTokenPayload: event.session.getAccessTokenPayload(),
        }),
        statusCode: 200,
    };
};
```

So in the above, I added a string like `__HIDDEN_TYPE_START__` before and `__HIDDEN_TYPE_END__` after the type definition. We can strip these strings away in each line when the ts files are being generated for code type checking, and then during generation of html, we can remove these strings and everything in between.

We should make sure that all these snippets are fully typed.

Reply:
done